packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -228,7 +229,10 @@ async function fetchLatestSnapshotCore(
                     storageToken,
                     snapshotOptions,
                     controller,
-                );
+                ).finally(() => {
```

Comment:
Does this cover the case when this async function is never called and the ODSP Driver is destroyed before that?

Reply:
I don't know, can we? It looks like the timer is set synchronously while its clearance asynchronously, so there is some time in between that could happen?

Reply:
The downloadSnapshot is called immediately. So, once it is called, then whether it fails/passes the timeout will be cleared. In cases where container is destroyed abruptly do the apps do that gracefully like calling close on the container? Like what I am asking is how does the driver/container knows if its destroyed or not?

Reply:
If an iframe were to get disposed, the `container.close()` method would get called.
For any hosts that don't use iframes, I think we can still assume that they'll call `container.close()`.

So I think Jatin's code looks good to me. It's hard to imagine a scenario where the `finally` codeblock wouldn't be getting executed...